pkg/plugin/args.go

```diff
@@ -0,0 +1,29 @@
+package plugin
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+)
+
+func LoadArgs(args string, container interface{}) error {
+	if args == "" {
+		return nil
+	}
+
+	js := `{`
+	pairs := strings.Split(args, ";")
+	for i, pair := range pairs {
+		kv := strings.Split(pair, "=")
+		if len(kv) != 2 {
+			return fmt.Errorf("No valid argument pair: %v", pair)
+		}
+		js += fmt.Sprintf(`"%v":"%v"`, kv[0], kv[1])
+		if i+1 < len(pairs) {
+			js += `,`
+		}
+	}
+	js += `}`
+
+	return json.Unmarshal([]byte(js), container)
```

Comment:
Bonus points if you can do this function without serializing to JSON first. Want to learn `reflect` api?